### PR TITLE
Add App-AutoScaler

### LINKED ISSUE #120: App Autoscaler
We'll be adding the App Autoscaler into 1.2, though it'll be considered experimental at this stage. This ties into an upstream project and will be off by default.

QA still need to test this but we can start documenting what we have whilst then pointing upstream.

Trello: https://trello.com/c/3wEpZBN4/641-8-implement-app-autoscaler-non-ha
Upstream: https://github.com/SUSE/scf/pull/1594

### PR DESCRIPTION
**Pull request by btat**
Resolves #120

### CODE REVIEW

File: xml/cap_admin_app_autoscaler.xml [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    count: 1
+  autoscaler_api:
+    count: 1
+  autoscaler_metrics:
+    count: 1
+  autoscaler_postgres:
+    count: 1
+</screen>
+  <para>
+   See <command>helm inspect suse/cf</command> for the full list of available values that can be used to configure the App-AutoScaler.
```
Comment (cjschroder) [inline xml/cap_admin_app_autoscaler.xml]: Does the Perl example in http://docserv.nue.suse.com/documents/CAP_1/cap-guides/single-html/#sec.cap.ha-defaults extract the necessary information for this?

**Comment by cjschroder** [comment]
so nice to see another feature documented \o/

File: xml/cap_admin_app_autoscaler.xml (hunk 1 above)
Comment (btat) [reply xml/cap_admin_app_autoscaler.xml]: Yep (see below), but the `see helm inspect suse/cf` is intended more for the other values available (which I'm not e.g. specify certs, timeouts, TTLs, etc) and not just enabling it.

```
helm inspect values suse/cf | \
perl -ne '/^sizing/..0 and do { print $.,":",$_ if /^ [a-z]/ || /high avail|scale|count/ }'
1049:    # The adapter instance group can scale between 1 and 65535 instances.
1050:    # For high availability it needs at least 2 instances.
1051:    count: 1
1091:    # The api-group instance group can scale between 1 and 65535 instances.
1092:    # For high availability it needs at least 2 instances.
1093:    count: 1
1105:  # The autoscaler-actors instance group contains the following jobs:
1113:  autoscaler_actors:
1120:    # The autoscaler-actors instance group can scale between 0 and 3 instances.
1121:    # For high availability it needs at least 2 instances.
1122:    count: 0
1134:  # The autoscaler-api instance group contains the following jobs:
1142:  autoscaler_api:
1149:    # The autoscaler-api instance group can scale between 0 and 3 instances.
1150:    # For high availability it needs at least 2 instances.
1151:    count: 0
1163:  # The autoscaler-metrics instance group contains the following jobs:
1171:  autoscaler_metrics:
1178:    # The autoscaler-metrics instance group can scale between 0 and 3 instances.
1179:    # For high availability it needs at least 2 instances.
1180:    count: 0
1192:  # The autoscaler-postgres instance group contains the following jobs:
1200:  autoscaler_postgres:
1207:    # The autoscaler-postgres instance group can scale between 0 and 3
1209:    # For high availability it needs at least 2 instances.
1210:    count: 0
```

Comment (cjschroder) [reply xml/cap_admin_app_autoscaler.xml]: ok!

**Pull request merged**